packages/blade/src/components/Drawer/_decisions/decisions.md

```diff
@@ -0,0 +1,178 @@
+
+<img width="500px" src="./2024-02-07-20-04-07.png" alt="Drawer Figma Skeleton">
+
+## Design
+
+- [Drawer - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=78667%3A66663&mode=dev)
+
+## Proposed API
+
+```jsx
+<Drawer
+  isOpen={}
+  onDismiss={() => {}}
+  showOverlay
+>
+  <DrawerHeader
+    title=""
+    subtitle=""
+    leading={<DrawerHeaderIcon />}
+    titleSuffix={<DrawerHeaderBadge />}
+    trailing={
+      <>
+        <Link />
+        <Button />
+      </>
+    }
+  />
+  <DrawerBody>
+    <Slot />
+  </DrawerBody>
+<Drawer>
+```
+
+<details>
+<summary><b>Full Usage Example</b></summary>
+
+```jsx
+const MyCuteDrawer = () => {
+  const [showDrawer, setShowDrawer] = React.useState(false);
+  return (
+    <Box>
+      <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+      <Drawer
+        isOpen={showDrawer}
+        onDismiss={() => {
+          setShowDrawer(false);
+        }}
+      >
+        <DrawerHeader
+          title="Announcements"
+        />
+        <DrawerBody>
+          <FTXAnnouncement />
+          <RazorpayOnePromotions />
+          <CatPictures />
+        </DrawerBody>
+      <Drawer>
+    </Box>
+  )
+}
+
+```
+
+</details>
+
+### Props
+
+```ts
+type DrawerProps = {
+  /**
+   * Controlled state of drawer open or not
+   */
+  isOpen: boolean;
+
+  /**
+   * Dismiss handler
+   */
+  onDismiss: () => void;
+
+  /**
+   * Show or hide overlay.
+   *
+   * Also decides if clicking outside on overlay closes the drawer or not
+   */
+  showOverlay?: boolean;
+
+  /**
+   * Initial focus reference element
+   */
+  initialFocusRef?: React.MutableRefObject<any>;
+
+  /**
+   * children node.
+   *
+   * Supports DrawerHeader and DrawerBody
+   */
+  children: React.ReactNode;
+
+  /**
+   * Override z-index of Drawer.
+   *
+   * @default 1002
+   */
+  zIndex?: number;
+
+  /**
+   *  Accessibility label for the drawer
+   */
+  accessibilityLabel?: string;
+};
+
+type DrawerHeaderProps = {
+  /**
+   * Title of the Drawer
+   */
+  title: string;
+
+  /**
+   * Subtitle of the Drawer
+   */
+  subtitle?: string;
+
+  /**
+   * Leading element
+   *
+   * DrawerHeaderIcon or DrawerHeaderAsset
+   */
+  leading?: ReactNode;
+
+  /**
+   * Title suffix element
+   *
+   * DrawerHeaderBadge
+   */
+  titleSuffix?: ReactNode;
+
+  /**
+   * Title trailing element
+   *
+   * Link, Button[]
+   */
+  trailing?: ReactNode;
+};
+```
+
+Other supporting wrapper components for trailing and leading space-
+
+- DrawerHeaderBadge
+- DrawerHeaderIcon
+- DrawerHeaderAsset
+
+_No alternate APIs were considered because Drawer is closer to Modal on overall meaning and API perspective so made sense to go with API that is closer to Modal, also all DS Drawer components I referenced have similar API_
+
+## Drawer Stacking
+
+- Only 2 Drawers can be stacked on top of each other
+- 2nd Drawer always has overlay independent of `showOverlay` prop
```

Comment:
So, the `showOverlay` prop is redundant for the 2nd drawer? Why so? If I don't want to show overlay in 1st drawer because important information is present on the left then that should hold for the 2nd drawer as well.

So I feel, if we are giving control of showing overlay it should be either on both drawers or none. WDYT?

Reply:
> So, the showOverlay prop is redundant for the 2nd drawer? Why so? If I don't want to show overlay in 1st drawer because important information is present on the left then that should hold for the 2nd drawer as well.

For example in X, 1st Drawer is used for showing information based on the interaction on left side. But 2nd Drawer does not have that usecase.

Plus overlay on 2nd Drawer is also there to denote the stacking. If there's 2 Drawers open without overlay, there is no direct way to tell user that there is another drawer behind this drawer.

Also in most cases, Drawer should always have overlay. There are just few exceptions for 1st Drawer which is why we need to provide overlay prop